src/main/kotlin/ecommerce/config/stripe/StripeClientConfiguration.kt

```diff
@@ -0,0 +1,26 @@
+package ecommerce.config.stripe
+
+import ecommerce.infrastructure.StripeClient
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.http.HttpHeaders
+import org.springframework.web.client.RestClient
+import org.springframework.web.client.support.RestClientAdapter
+import org.springframework.web.service.invoker.HttpServiceProxyFactory
+
+@Configuration
+class StripeClientConfiguration {
+    @Bean
+    fun stripeClient(stripeProperties: StripeProperties): StripeClient {
+        val restClient =
+            RestClient.builder()
+                .baseUrl("https://api.stripe.com")
+                .defaultHeader(HttpHeaders.AUTHORIZATION, "Bearer ${stripeProperties.secretKey}")
+                .build()
+
```

Comment:
```suggestion
@Configuration
class StripeClientConfiguration {
    @Bean
     fun stripeClient(stripeProperties: StripeProperties): StripeClient {
        val requestFactory = SimpleClientHttpRequestFactory().apply {
            setConnectTimeout(Duration.ofMillis(500))
            setReadTimeout(Duration.ofSeconds(10))
        }

        val restClient =
            RestClient.builder()
                .requestFactory(requestFactory)
                .baseUrl("https://api.stripe.com")
                .defaultHeader(HttpHeaders.AUTHORIZATION, "Bearer ${stripeProperties.secretKey}")
                .build()


```

Reply:
> I found an alternative approach using only the config and created a test in the end-to-end admin suite for the /slow endpoint. Please let me know if this is the best approach or if there are better options.

Testing timeouts is never easy. A common approach is to set up a separate API and run a server just for testing. You did a great job.

If the controller is only used for testing, another option is to place it under the test package instead of main. 😉